api/src/main/java/com/cloud/storage/Snapshot.java

```diff
@@ -26,7 +26,7 @@
 
 public interface Snapshot extends ControlledEntity, Identity, InternalIdentity, StateObject<Snapshot.State> {
     public enum Type {
-        MANUAL, RECURRING, TEMPLATE, HOURLY, DAILY, WEEKLY, MONTHLY;
+        MANUAL, RECURRING, TEMPLATE, HOURLY, DAILY, WEEKLY, MONTHLY, INTERNAL;
```

Comment:
Better to add a comment to indicate any new type can be added after INTERNAL (so that the internal enum value [7] doesn't change)

Reply:
maybe this will work but as we have now two fixed values, it is maybe time to introduce a construcor with values to use?